src/main/java/com/shopping_app/shoppingApp/service/AuthenticationService.java

```diff
@@ -0,0 +1,19 @@
+package com.shopping_app.shoppingApp.service;
+
+import com.shopping_app.shoppingApp.Exceptions.BaseException;
+import com.shopping_app.shoppingApp.utils.UserPrincipal;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthenticationService {
+
+    public boolean isAuthenticate(String userId) {
+        UserPrincipal principal = (UserPrincipal) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        if (!userId.equals(String.valueOf(principal.getId()))) {
+            throw new BaseException("User does not have access to this resource", HttpStatus.FORBIDDEN);
```

Comment:
You don't need to throw an exception here, just returning a boolean should be fine